api/src/main/java/org/openmrs/util/EnversAuditTableInitializer.java

```diff
@@ -0,0 +1,162 @@
+import org.hibernate.tool.schema.spi.ExceptionHandler;
+import org.hibernate.tool.schema.spi.ExecutionOptions;
+import org.hibernate.tool.schema.spi.SchemaFilter;
+import org.hibernate.tool.schema.spi.SchemaManagementTool;
+import org.hibernate.tool.schema.spi.SchemaMigrator;
+import org.hibernate.tool.schema.spi.ScriptTargetOutput;
+import org.hibernate.tool.schema.spi.TargetDescriptor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Initializes Hibernate Envers audit tables when auditing is enabled. This class is responsible for
+ * conditionally creating audit tables only when hibernate.integration.envers.enabled=true.
+ */
+public class EnversAuditTableInitializer {
+	
+	private static final Logger log = LoggerFactory.getLogger(EnversAuditTableInitializer.class);
+	
+	/**
+	 * Checks if Envers is enabled and creates/updates audit tables as needed. This will Create or
+	 * Update audit tables if they don't exist - Update existing audit tables if the schema has
+	 * changed
+	 * 
+	 * @param metadata Hibernate metadata containing entity mappings
+	 * @param hibernateProperties properties containing Envers configuration
+	 * @param serviceRegistry Hibernate service registry
+	 */
+	public static void initialize(Metadata metadata, Properties hibernateProperties,
+								  ServiceRegistry serviceRegistry) {
+		
+		if (!isEnversEnabled(hibernateProperties)) {
+			log.debug("Hibernate Envers is not enabled. Skipping audit table initialization.");
+			return;
+		}
+		
+		try {
+			updateAuditTables(metadata, hibernateProperties, serviceRegistry);
+			log.info("Envers audit table initialization completed successfully.");
+		}
+		catch (Exception e) {
+			log.error("Failed to initialize Envers audit tables. ", e);
+		}
+	}
+	
+	/**
+	 * Checks if Hibernate Envers is enabled in the configuration.
+	 * 
+	 * @param properties Hibernate properties
+	 * @return true if Envers is enabled, false otherwise
+	 */
+	private static boolean isEnversEnabled(Properties properties) {
+		String enversEnabled = properties.getProperty("hibernate.integration.envers.enabled");
+		return "true".equalsIgnoreCase(enversEnabled);
+	}
+	
+	/**
+	 * Creates or updates audit tables using Hibernate's {@link SchemaMigrator}. This method filters
+	 * to only process audit tables.
+	 * 
+	 * @param metadata Hibernate metadata containing entity mappings (includes Envers audit
+	 *            entities)
+	 * @param hibernateProperties Hibernate configuration properties
+	 * @param serviceRegistry Hibernate service registry
+	 * @throws Exception if table creation/update fails
+	 */
+	private static void updateAuditTables(Metadata metadata, Properties hibernateProperties,
+	        ServiceRegistry serviceRegistry) throws Exception {
+		String auditTablePrefix = hibernateProperties.getProperty("org.hibernate.envers.audit_table_prefix", "");
+		String auditTableSuffix = hibernateProperties.getProperty("org.hibernate.envers.audit_table_suffix", "_aud");
```

Comment:
The main thing I might ask here is that we name the audit tables `_audit`. It's more consistent with how we name other DB tables and this isn't MUMPS where tables names are (IIRC) something like 10 characters.